### ISSUE TITLE
ModelicaStandardTables should check range of columns

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1816]
**Reported by henrikt on 3 Nov 2015 10:13 UTC**
There should be an error message for using values in `columns` that are outside the range of available columns.

For example, `ModelicaStandardTables_CombiTable1D_read` should have something like
```
        for (size_t i = 0; i < tableID->nCols; ++i) {
          if (tableID->cols[i] < 0 || tableID->nCol <= tableID->cols[i]) {
            ModelicaFormatError("In table %s in %s, column index out of range: %d\n",
              tableID->tableName, tableID->fileName, tableID->cols[i]);
          }
        }
```
after the table dimensions have been initialized.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1816

**Comment by modelica-trac-importer** [comment]
**Modified by beutlich on 3 Nov 2015 13:51 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 3 Nov 2015 15:02 UTC**
Resolved by 8767e6a1ea63514c89a0a87645beea31a44e4c5a.

**Comment by modelica-trac-importer** [comment]
**Changelog modified by beutlich on 3 Nov 2015 15:02 UTC**
Modelica Standard Tables: Added range checks for column indices of CombiTimeTable and CombiTable1D
